### PR TITLE
Deprecate history task processor 

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Deprecate history task processor
- Replace it with new task executable, scheduler, rescheduler, priority assigner implementation
- Update queue ack manager & processor to use above new components
- Noop priority assigner implementation
- Clean up unused code and configs

**Why?**
- Host level worker pool

**How did you test it?**
- Unit test, integration test, canary.

**Potential risks**
- In the worst case, task processing may be blocked and task may be lost.

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/configs/task.go [hunk 1]
```diff
@@ -64,6 +68,34 @@ func ConvertWeightsToDynamicConfigValue(
...
 }
 
+func ConvertDynamicConfigValueToWeights(
+	weightsFromDC map[string]interface{},
+) map[int]int {
+	weights := make(map[int]int)
+	for key, value := range weightsFromDC {
+		intKey, err := strconv.Atoi(strings.TrimSpace(key))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to convert dynamicconfig map value to int map: %v is not integer", key))
```
Comment (wxing1292) [inline service/history/configs/task.go:78]: if operator make a mistake, will this crush the service?

Comment (wxing1292) [reply service/history/configs/task.go:78]: and do we really need dynamic weight?

File: service/history/historyEngineInterfaces.go [hunk 2]
```diff
@@ -57,43 +58,29 @@ type (
 
 	queueAckMgr interface {
 		getFinishedChan() <-chan struct{}
-		readQueueTasks() ([]tasks.Task, bool, error)
-		completeQueueTask(taskID int64)
+		readQueueTasks() ([]queues.Executable, bool, error)
```
Comment (wxing1292) [inline service/history/historyEngineInterfaces.go:61]: just want to confirm, this `readQueueTasks() ([]queues.Executable, bool, error)` is temporary?

File: service/history/configs/task.go (hunk 1 above)
Comment (yycptt) [reply service/history/configs/task.go:78]: Good point. let me emit a log and fallback to default weight if dc value can't be parsed.

I think dynamic config is still necessary as it's hard to estimate the real effect of those weights without looking at production metrics. Existing iwrr scheduler implementation don't support change the weight on the fly. I will modify it later to support it and make the weights real dynamic.

File: service/history/historyEngineInterfaces.go (hunk 2 above)
Comment (yycptt) [reply service/history/historyEngineInterfaces.go:61]: Yes. All interfaces defined in this file will be removed after multicursor queue is done.

File: service/history/timerQueueProcessorBase.go [hunk 3]
```diff
@@ -157,15 +151,13 @@ func (t *timerQueueProcessorBase) Stop() {
 
 	t.timerGate.Close()
 	close(t.shutdownCh)
-	t.retryTasks()
+	// t.retryTasks()
```
Comment (wxing1292) [inline service/history/timerQueueProcessorBase.go:154]: remove this line?

File: service/history/timerQueueProcessorBase.go [hunk 4]
```diff
@@ -340,36 +345,53 @@ func (t *timerQueueProcessorBase) readAndFanoutTimerTasks() (*time.Time, error)
...
-func (t *timerQueueProcessorBase) isPriorityTaskProcessorEnabled() bool {
-	return t.taskProcessor == nil
+	submitted, err := t.executableScheduler.TrySubmit(executable)
+	if err != nil {
+		t.logger.Error("Failed to submit task", tag.Error(err))
+	}
+
+	if !submitted {
+		executable.Reschedule()
+	}
```
Comment (wxing1292) [inline service/history/timerQueueProcessorBase.go:375]:
```go
	if err != nil {
		t.logger.Error("Failed to submit task", tag.Error(err))
		executable.Reschedule()
	} else if !submitted {
		executable.Reschedule()
	}
```

File: service/history/queueProcessor.go [hunk 5]
```diff
@@ -271,31 +263,40 @@ func (p *queueProcessorBase) processBatch() {
...
-	p.ackMgr.completeQueueTask(task.GetKey().TaskID)
+
+	submitted, err := p.executableScheduler.TrySubmit(executable)
+	if err != nil {
+		p.logger.Error("Failed to submit task", tag.Error(err))
+	}
+
+	if !submitted {
+		executable.Reschedule()
+	}
```
Comment (wxing1292) [inline service/history/queueProcessor.go:301]:
```go
	submitted, err := p.executableScheduler.TrySubmit(executable)
	if err != nil {
		p.logger.Error("Failed to submit task", tag.Error(err))
		executable.Reschedule()
	} else if !submitted {
		executable.Reschedule()
	}
```

File: service/history/configs/task.go (hunk 1 above)
Comment (wxing1292) [reply service/history/configs/task.go:78]: >I will modify it later to support it and make the weights real dynamic.

before making everything dynamic, let us sync, we may need to evaluate "making everything dynamic all the time" vs "ability to reload according to signal"

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**